packages/filesystem/src/browser/location/location-renderer.tsx

```diff
@@ -42,14 +48,30 @@ export class LocationListRenderer extends ReactRenderer {
     protected readonly handleLocationChanged = (e: React.ChangeEvent<HTMLSelectElement>) => this.onLocationChanged(e);
     protected doRender(): React.ReactNode {
         const location = this.service.location;
-        const locations = !!location ? location.allLocations : [];
+        const locations = (!!location ? location.allLocations : []).map(uri => ({ uri }));
```

Comment:
could you please extract L50 to L66 to a protected method `computeLocations` (name it as you wish), which could make filtering easier for extenders.